client/src/views/ActiveUsersPage.vue

```diff
@@ -79,36 +172,153 @@ import {
 import ActionBar from '@/components/ActionBar.vue';
+import { MockUser, getMockUsers } from '@/common/mocks';
+import { Ref } from 'vue';
+import { onMounted } from '@vue/runtime-core';
 
 const displayView = ref(DisplayState.List);
-const userList = ['User1', 'User2', 'User3'];
+const userList: Ref<MockUser[]> = ref([]);
+const userListItemRefs: Ref<typeof UserListItem[]> = ref([]);
+const userGridItemRefs: Ref<typeof UserCard[]> = ref([]);
+const allUsersSelected = ref(false);
+const interminateState = ref(false);
 
 const filteredUsers = computed(() => {
-  return userList;
+  const revokedUsers = userList.value.filter((user) => {
+    return user.revoked === false;
+  });
+  return revokedUsers;
+});
+
+const selectedUsersCount = computed(() => {
+  if (displayView.value === DisplayState.List) {
+    return userListItemRefs.value.filter((item) => item.isSelected).length;
+  } else {
+    return userGridItemRefs.value.filter((item) => item.isSelected).length;
+  }
 });
 
 function inviteUser(): void {
   console.log('Invite user clicked');
 }
+
+function viewCommonWorkspace(): void {
+  console.log('View common workspace clicked');
+}
+
+function onUserSelect(_user: MockUser, _selected: boolean): void {
+  console.log(selectedUsersCount.value);
+
+  if (selectedUsersCount.value === 0) {
+    allUsersSelected.value = false;
+    selectAllUsers(false);
+  }
+  // check global checkbox if all users are selected
+  if (selectedUsersCount.value === userList.value.length) {
+    allUsersSelected.value = true;
+  } else {
+    allUsersSelected.value = false;
+  }
+
+  if (selectedUsersCount.value === 0 || selectedUsersCount.value === userList.value.length) {
+    interminateState.value = false;
+  } else {
+    console.log('No users selected');
+    interminateState.value = true;
+  }
+}
+
+function selectAllUsers(checked: boolean): void {
+  if (displayView.value === DisplayState.List) {
+    for (const item of userListItemRefs.value || []) {
+      item.isSelected = checked;
+      if (checked) {
+        item.showCheckbox = true;
+      } else {
+        item.showCheckbox = false;
+      }
+    }
+  } else {
+    for (const item of userGridItemRefs.value || []) {
+      item.isSelected = checked;
+      if (checked) {
+        item.showCheckbox = true;
+      } else {
+        item.showCheckbox = false;
+      }
+    }
+  }
+}
+
+function revokeUser(): void {
+  console.log('Revoke user clicked');
+}
+
+function details(): void {
+  console.log('Details user clicked');
+}
+
+async function openUserContextMenu(event: Event, user: MockUser): Promise<void> {
+  const popover = await popoverController
+    .create({
+      component: UserContextMenu,
+      cssClass: 'user-context-menu',
+      event: event,
+      translucent: true,
+      showBackdrop: false,
+      dismissOnSelect: true,
+      reference: 'event',
+    });
+  await popover.present();
+
+  const { data } = await popover.onDidDismiss();
```

Comment:
Handle the case where `data` is undefined (click outside of the menu for example).

Reply:
Fix with your recommandation
```
if (!data) {
  return;
}
```